src/utils/init/config-github.js

```diff
@@ -6,34 +6,55 @@ const { getGitHubToken: ghauth } = require('../gh-auth')
 
 const { createDeployKey, formatErrorMessage, getBuildSettings, saveNetlifyToml, setupSite } = require('./utils')
 
+/**
+ * @typedef Token
+ * @type {object}
+ * @property {string} user - The username that is associated with the token
+ * @property {string} token - The actual token value.
+ * @property {string} provider - The Provider where the token is associated with ('github').
+ */
+
 const formatRepoAndOwner = ({ repoName, repoOwner }) => ({
   name: chalk.magenta(repoName),
   owner: chalk.magenta(repoOwner),
 })
 
 const PAGE_SIZE = 100
 
-const isValidToken = (token) => token && token.user && token.token
-
+/**
+ * Get a valid github token
+ * @returns {string}
+ */
 const getGitHubToken = async ({ globalConfig }) => {
   const userId = globalConfig.get('userId')
+
+  /** @type {Token} */
   const githubToken = globalConfig.get(`users.${userId}.auth.github`)
 
-  if (isValidToken(githubToken)) {
-    return githubToken.token
+  if (githubToken && githubToken.user && githubToken.token) {
+    try {
+      const octokit = getGitHubClient(githubToken.token)
+      const { status } = await octokit.rest.users.getAuthenticated()
+      if (status < 400) {
+        return githubToken.token
+      }
+    } catch {}
```

Comment:
Should we add a log message telling the user their token is expired?

Reply:
Will do! Good point :)